zenodo/modules/records/query.py

```diff
@@ -30,11 +30,15 @@
 from flask import request
 from invenio_records_rest.query import es_search_factory as _es_search_factory
 
-
 def apply_version_filters(search, urlkwargs):
     """Apply record version filters to search."""
     if request and 'all_versions' in request.values:
-        urlkwargs.add('all_versions', None)
+        if((str(request.values['all_versions']) == "1") or
+                str(request.values['all_versions'].lower()) == "true"):
+            urlkwargs.add('all_versions', "true")
+
+        else:
+            urlkwargs.add('all_versions', str(request.values['all_versions']))
```

Comment:
Since we assume this is a "falsey" value for the `all_versions` parameter, you have to also apply the search filter below:

```suggestion
            urlkwargs.add('all_versions', str(request.values['all_versions']))
            search = search.filter(Q('term', **{'relations.version.is_last': True}))
```

...or even better store it in a variable and replace the `else` clause below with an `if ...`